src/components/Common/UI/Button/ButtonTypes.ts

```diff
@@ -0,0 +1,31 @@
+import type { Ref, ButtonHTMLAttributes, ReactNode, FocusEvent } from 'react'
+
+// Define event handler types that are compatible with both HTML elements and React Aria
+type ButtonFocusHandler = (e: FocusEvent) => void
+
+export interface ButtonProps
+  extends Pick<
+    ButtonHTMLAttributes<HTMLButtonElement>,
+    | 'name'
+    | 'id'
+    | 'className'
+    | 'type'
+    | 'onClick'
+    | 'onKeyDown'
+    | 'onKeyUp'
+    | 'aria-label'
+    | 'aria-labelledby'
+    | 'aria-describedby'
+    | 'form'
+    | 'title'
+    | 'tabIndex'
+  > {
+  buttonRef?: Ref<HTMLButtonElement>
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'link' | 'icon'
```

Comment:
I'm wondering if these should actually be broken out into multiple button components. Ex. feeling `LinkButton` should be a separate component. It seems so different from the rest of the button styles and may have additional accessibility considerations to account for. To a lesser degree i'd still push for icon to be a distinct component, ex. IconButton, as well. That would be so these could take care of their own clear dedicated API concerns rather than having everything needing to be combined with the button. From experience, icon buttons end up with different sizing considerations and labeling considerations (since they strictly use visually hidden text)

Reply:
I'm game to do this, i'm thinking I might want to do it in a different story just so we're not changing a ton of files. 

That said I just did research and we have zero uses of LinkButton in our app only our stories. We have 2 uses of Icon button in our app in Hamburger component + Pagination control. 

Given the low usage, would you be ok with waiting to do this in another story, or does it low usage for both components (And only being used inside our component library itself) warrant waiting?

Reply:
Yes i think separate story is preferable! we'll def want to prioritize it before the api for adapters goes live but no need to do everything all at once here

interesting that link button is not used. If we can get away with that not existing and just removing it that'd be much less confusing for consumers :D maybe we could chat with @aaronlee777 about it